splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/upgrade/SpliceCatalogUpgradeScripts.java

```diff
@@ -33,79 +30,121 @@
     protected static final Logger LOG=Logger.getLogger(SpliceCatalogUpgradeScripts.class);
 
     SpliceDataDictionary sdd;
-    Splice_DD_Version catalogVersion;
     TransactionController tc;
-    TreeMap<Splice_DD_Version, UpgradeScript> scripts;
-    Comparator<Splice_DD_Version> ddComparator;
-    
-    public SpliceCatalogUpgradeScripts(SpliceDataDictionary sdd,Splice_DD_Version catalogVersion,TransactionController tc){
+
+    List<VersionAndUpgrade> scripts;
+
+    public static class VersionAndUpgrade {
+        public Splice_DD_Version version;
+        public UpgradeScript script;
+
+        public VersionAndUpgrade(Splice_DD_Version version,UpgradeScript script) {
+            this.version = version;
+            this.script = script;
+        }
+    }
+
+    public void addUpgradeScript(Splice_DD_Version version, int sprint, UpgradeScript script)
+    {
+        scripts.add(new VersionAndUpgrade(new Splice_DD_Version(sdd, version.getMajorVersionNumber(),
+                version.getMinorVersionNumber(), version.getPatchVersionNumber(), sprint), script));
+    }
+
+    // master / branch-3.0 / branch-3.1
+    // baseVersion1 = 2.8.0 / 2.8.0 / 2.8.0
+    // baseVersion2 = 3.1.0 / 3.1.0 / 3.0.0 // 3. fork
+    // baseVersion3 = 3.1.0 / 3.1.0 / 3.0.1 // hickup on branch 3.0
+    // baseVersion4 = 3.2.0 / 3.1.0 / 3.0.1 // 3.2 fork
```

Comment:
Can you add two lines here explaining what the table means? It took me a while to understand

Reply:
yes, good point